pkg/authorization/middleware.go

```diff
@@ -71,7 +71,12 @@ func (m *Middleware) WrapHandler(handler func(ctx context.Context, w http.Respon
 			userAuthNMethod = "TLS"
 		}
 
-		authCtx := NewCtx(plugins, user, userAuthNMethod, r.Method, r.RequestURI)
+		remoteAddr := r.RemoteAddr
+		if ip := r.Header.Get("X-Forwarded-For"); ip != "" {
```

Comment:
This is more complicated that this: you can handle this but also `Forwarded`. ;)

Reply:
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded for how those headers look; looks like we need some additional parsing for that